rewrite-core/src/main/java/org/openrewrite/ExcludeFileFromGitignore.java

```diff
@@ -268,4 +220,190 @@ private IgnoreNode.MatchResult isIgnored(IgnoreNode ignoreNode, String path) {
             return CHECK_PARENT;
         }
     }
+
+    private static class IgnoreRule {
```

Comment:
I do wonder if we should be applying these changes in our fork of jGit instead, but I think that given we are also making changes to how we want the changes written this is fine